Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -3962,9 +3962,10 @@ void FrameLoader::loadItem(HistoryItem& item, HistoryItem* fromItem, FrameLoadTy
     // If we're continuing this history navigation in a new process, then doing a same document navigation never makes sense.
     ASSERT(!sameDocumentNavigation || shouldTreatAsContinuingLoad == ShouldTreatAsContinuingLoad::No);
 
-    if (sameDocumentNavigation)
+    if (sameDocumentNavigation) {
+        m_loadType = loadType;
         loadSameDocumentItem(item);
```

Comment:
I wonder if `loadSameDocumentItem` should take a '`loadType`' argument, just like `loadDifferentDocumentItem`.

Reply:
But then we have to initialize m_loadType at some point. The question is, in which function is it best to initialize m_loadType. I wasn't sure so I just initialized it there.

In the other code path, they're initializing m_loadType *after* navigation policy but there is no navigation policy here it seems.

Reply:
Ah, okay. This is fine as-is.